src/odemis/gui/cont/tabs.py

```diff
@@ -1870,6 +1870,22 @@ def display_new_data(self, filename, data, extend=False):
         # TODO: Move viewport related code to ViewPortController
         # TODO: to support multiple (types of) streams (eg, AR+Spec+Spec), do
         # this every time the streams are hidden/displayed/removed.
+
+        # handle display when both data types are present.
+        if spec_streams and ar_streams:
+            dlg = wx.MessageDialog(None, "The file %s contains both spectrum and AR data. Which data type should be displayed?" % (filename,),
```

Comment:
Drop the filename from the message. It's typically long and confusing, and normally the user knows very well which file you're talking about (as he/she has just selected it).